YakimS/assignment4/IslandsCounter.java

```diff
@@ -0,0 +1,55 @@
+package com.shaya.CodeUAss4Package;
+
+import java.util.Arrays;
+
+public class IslandsCounter {
+    /**
+     *  Counts the number of islands in a 2-dimensional array,
+     *  which each tile of it is either land(=true) or water(=false).

```

Comment:
Typically details about contents of parameters is done  as `@param` line rather than in the main method description. This makes it possible for IDE to parse comment and show description of parameter as a tooltip when you hover over it in code, which is really convenient.